### PR TITLE
Handle 0 maxSize in `DefaultConnectionPool` and allow passing 0 to `MongoClientOptions.Builder.maxSize`

### PR DESCRIPTION
**Pull request by stIncMale**
See also JAVA-4160 and [CMAP Connection Pool Options]( https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#connection-pool-options-1).

JAVA-4216

### CODE REVIEW

File: driver-core/src/main/com/mongodb/connection/ConnectionPoolSettings.java [hunk 1]
```diff
@@ -410,7 +410,7 @@ public String toString() {
     }
 
     ConnectionPoolSettings(final Builder builder) {
-        isTrue("maxSize > 0", builder.maxSize > 0);
+        isTrue("maxSize >= 0", builder.maxSize >= 0);
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/connection/ConnectionPoolSettings.java:413]: Consider adding a unit test in `ConnectionPoolSettingsSpecification`

**Review by jyemin — CHANGES_REQUESTED** [review]
LGTM after unit test added

File: driver-core/src/main/com/mongodb/connection/ConnectionPoolSettings.java (hunk 1 above)
Comment (stIncMale) [reply driver-core/src/main/com/mongodb/connection/ConnectionPoolSettings.java:413]: Done.

**Comment by stIncMale** [comment]
This [test failure](https://spruce.mongodb.com/task/mongo_java_driver_tests_load_balancer_secure__version~latest_os~ubuntu_topology~sharded_cluster_auth~auth_ssl~ssl_jdk~jdk11_load_balancer_test_patch_2b621b3c761e2b145cada33f2d65beb58c9615f7_6101d8587742ae0e9b09e955_21_07_28_22_21_13/tests?execution=0&sortBy=STATUS&sortDir=ASC&statuses=fail) is the same as reported in JAVA-4259.

**Review by jyemin — APPROVED** [review]
LGTM

**Review by rozza — APPROVED** [review]
LGTM

**Pull request merged**